# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/builtin/RegExp.cpp
# Commit: e7d230029514
# Full Hash: e7d2300295143bf951d8bdecd7619ea78722bb80
# Author: Iain Ireland <iireland@mozilla.com>
# Date: 2020-05-22 21:38:53
# Regressor Bug: 1362154
# File Overlap Count: 1
# Description:
#   Bug 1362154: Part 2: Avoid rooting hazard in CheckPatternSyntax r=mgaudet
#   
#   In debug, CheckPatternSyntax holds an unrooted RegExpShared across a call to the regexp parser. This is the only remaining obstacle to allocating GC things while parsing regexps.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D76034
# ==============================================================================

diff -r 7f7a204fa024 -r e7d230029514 js/src/builtin/RegExp.cpp
--- a/js/src/builtin/RegExp.cpp	Wed May 20 17:30:56 2020 +0000
+++ b/js/src/builtin/RegExp.cpp	Wed May 20 21:09:36 2020 +0000
@@ -209,7 +209,8 @@
   // If we already have a RegExpShared for this pattern/flags, we can
   // avoid the much slower CheckPatternSyntaxSlow call.
 
-  if (RegExpShared* shared = cx->zone()->regExps().maybeGet(pattern, flags)) {
+  RootedRegExpShared shared(cx, cx->zone()->regExps().maybeGet(pattern, flags));
+  if (shared) {
 #ifdef DEBUG
     // Assert the pattern is valid.
     if (!CheckPatternSyntaxSlow(cx, pattern, flags)) {
